performance/plugins/performanceTestItem.ts

```diff
@@ -26,71 +34,162 @@ export class performanceTestItem implements TestItem {
-    const mga = await getMangata(nodes[0]!);
+
+  enqueued: Promise<[number, number][]> = new Promise<[number, number][]>(
+    (resolve) => {
+      resolve([]);
+    }
+  );
+  executed: Promise<[number, number][]> = new Promise<[number, number][]>(
+    (resolve) => {
+      resolve([]);
+    }
+  );
+  pending: Promise<[number, number][]> = new Promise<[number, number][]>(
+    (resolve) => {
+      resolve([]);
+    }
+  );
+  ipc: any;
+
+  async arrange(testParams: TestParams): Promise<boolean> {
+    const mga = await getMangata(testParams.nodes[0]!);
     const api = await mga.getApi();
-    this.listeners.push(captureEvents(logFile, api!));
-    this.listeners.push(pendingExtrinsics(logFile, api!));
+    const { sudo } = getEnvironmentRequiredVars();
+    const keyring = new Keyring({ type: "sr25519" });
+    const sudoKeyringPair = keyring.createFromUri(sudo);
+    const nonce = await api.rpc.system.accountNextIndex(
+      sudoKeyringPair.address
+    );
+
+    const ipc = require("node-ipc").default;
+    ipc.config.id = "nonceManager";
+    ipc.config.retry = 1500;
+    ipc.config.silent = false;
+    ipc.config.sync = true;
+    ipc.serve(function () {
+      ipc.server.on("getNonce", (data: any, socket: any) => {
+        console.info("serving nonce" + data.id + " " + nonce);
+        ipc.server.emit(socket, "nonce-" + data.id, nonce.toNumber());
+        nonce.iaddn(1);
+      });
+    });
+
+    ipc.server.start();
+    this.ipc = ipc;
 
     return true;
   }
+
   async act(testParams: TestParams): Promise<boolean> {
-    testLog.getLog().info("Act" + testParams);
+    const mga = await getMangata(testParams.nodes[0]!);
+    const api = await mga.getApi();
+    this.executed = trackExecutedExtrinsics(api, testParams.duration);
+    this.enqueued = trackEnqueuedExtrinsics(api, testParams.duration);
+    this.pending = trackPendingExtrinsics(api, testParams.duration);
     return true;
   }
-  async expect(nodes: string[]): Promise<boolean> {
-    //wait for not Txs pending + 10 blocks.
-    const mga = await getMangata(nodes[0]!);
-    let pendingExtr: any[] = [];
-    do {
-      const api = await mga.getApi();
-      pendingExtr = await api.rpc.author.pendingExtrinsics();
-    } while (pendingExtr.length > 0);
-    console.info(`Done waiting Txs!`);
-    console.info(`Stopping listeners....`);
-    this.listeners.forEach(async (p) => (await p).cancel());
-    console.info(`...Stopped`);
+
+  async expect(testParams: TestParams): Promise<boolean> {
+    let [executed, enqueued, pending] = await Promise.all([
+      this.executed,
+      this.enqueued,
+      this.pending,
+    ]);
+
+    writeToFile("executed.txt", executed);
+    writeToFile("enqueued.txt", enqueued);
+    writeToFile("pending.txt", pending);
+    generateHtmlReport("report.html", enqueued, executed, pending);
+
+    // ignore first few blocks
+    executed = executed.slice(5);
+    enqueued = enqueued.slice(5);
+    pending = pending.slice(5);
+
```

Comment:
ignore first few blocks when node is warming up (only for quantile calculations)